Build fails with `server.cc:284:9: error: ‘UpnpSetWebServerCorsString’ was not declared in this scope; did you mean ‘UpnpSetWebServerRootDir’?`. [GCC 14, libnpupnp 6.3.1.]

Ahoj,

### Short:

Build fails with `server.cc:284:9: error: ‘UpnpSetWebServerCorsString’ was not declared in this scope; did you mean ‘UpnpSetWebServerRootDir’?`:  
```
[...]
[ 50%] Building CXX object CMakeFiles/libgerbera.dir/src/server.cc.o
/var/cache/makepkg/build/gerbera-nosystemd-git/src/gerbera/src/server.cc: In member function ‘void Server::run()’:
/var/cache/makepkg/build/gerbera-nosystemd-git/src/gerbera/src/server.cc:284:9: error: ‘UpnpSetWebServerCorsString’ was not declared in this scope; did you mean ‘UpnpSetWebServerRootDir’?
  284 |         UpnpSetWebServerCorsString(fmt::format("{}", fmt::join(corsHosts, " ")).c_str());
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~
      |         UpnpSetWebServerRootDir
make[2]: *** [CMakeFiles/libgerbera.dir/build.make:1378: CMakeFiles/libgerbera.dir/src/server.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:207: CMakeFiles/libgerbera.dir/all] Error 2
[...]
```

I use GCC 14.2.1 and libnpupnp 6.1.3.


### Long:

I try to build the latest git checkout (`git describe --tags`: `v2.2.0-53-g52c4f06b9`) of this software as follows:

```
cmake . \
  -DCMAKE_BUILD_TYPE=Release \
  -DCMAKE_INSTALL_PREFIX=/usr \
  -DCXX_FILESYSTEM_HAVE_FS=ON \
  -DEXIF_ROOT_DIR=/usr \
  -DSTATIC_LIBUPNP=OFF \
  -DWITH_AVCODEC=ON \
  -DWITH_CURL=ON \
  -DWITH_DEBUG=ON \
  -DWITH_EXIF=ON \
  -DWITH_EXIV2=ON \
  -DWITH_FFMPEGTHUMBNAILER=ON \
  -DWITH_INOTIFY=ON \
  -DWITH_JS=ON \
  -DWITH_LASTFM=OFF \
  -DWITH_MAGIC=ON \
  -DWITH_MATROSKA=ON \
  -DWITH_MYSQL=OFF \
  -DWITH_NPUPNP=ON \
  -DWITH_SYSTEMD=OFF \
  -DWITH_TAGLIB=ON \
  -DWITH_TESTS=ON

make -j1
```

The `make` call fails with:  
```
[...]
[ 50%] Building CXX object CMakeFiles/libgerbera.dir/src/request_handler/url_request_handler.cc.o
[ 50%] Building CXX object CMakeFiles/libgerbera.dir/src/server.cc.o
/var/cache/makepkg/build/gerbera-nosystemd-git/src/gerbera/src/server.cc: In member function ‘void Server::run()’:
/var/cache/makepkg/build/gerbera-nosystemd-git/src/gerbera/src/server.cc:284:9: error: ‘UpnpSetWebServerCorsString’ was not declared in this scope; did you mean ‘UpnpSetWebServerRootDir’?
  284 |         UpnpSetWebServerCorsString(fmt::format("{}", fmt::join(corsHosts, " ")).c_str());
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~
      |         UpnpSetWebServerRootDir
make[2]: *** [CMakeFiles/libgerbera.dir/build.make:1378: CMakeFiles/libgerbera.dir/src/server.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:207: CMakeFiles/libgerbera.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
```


### More context:

* Operating System: [Artix GNU/Linux](http://artixlinux.org/) (rolling release)
* `gcc --version | head -n1`: `gcc (GCC) 14.2.1 20240805`
* `libnpupnp` version: 6.1.3
* Output of the `cmake` run:  
  ```
  -- The CXX compiler identification is GNU 14.2.1
  -- The C compiler identification is GNU 14.2.1
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Check for working CXX compiler: /usr/bin/c++ - skipped
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: /usr/bin/cc - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Gerbera Version: v2.2.0-53-g52c4f06b9
  -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
  -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
  -- Found Threads: TRUE
  -- Looking for __atomic_fetch_add_4 in atomic
  -- Looking for __atomic_fetch_add_4 in atomic - found
  -- Looking for C++ include filesystem
  -- Looking for C++ include filesystem - found
  -- Performing Test CXX_FILESYSTEM_NO_LINK_NEEDED
  -- Performing Test CXX_FILESYSTEM_NO_LINK_NEEDED - Success
  -- Performing Test Iconv_IS_BUILT_IN
  -- Performing Test Iconv_IS_BUILT_IN - Success
  -- Found Iconv: built in to C library
  -- Checking prototype iconv for ICONV_CONST
  -- Checking prototype iconv for ICONV_CONST - False
  -- Looking for BSD native UUID
  -- Looking for libuuid
  -- Looking for libuuid - found
  -- Found UUID: TRUE
  -- Looking for native LFS support
  -- Looking for native LFS support - found
  -- Found LFS: TRUE
  -- found npupnp include at /usr/include/npupnp
  -- Found libnpupnp: /usr/lib/libnpupnp.so (found version 6.1.3)
  -- Found spdlog: /usr/lib/cmake/spdlog/spdlogConfig.cmake (found suitable version "1.14.1", minimum required is "1.8.1")
  -- Found pugixml: /usr/lib/libpugixml.so
  -- Found SQLite3: /usr/include (found suitable version "3.46.1", minimum required is "3.7.11")
  -- Found duktape: /usr/lib/libduktape.so
  -- Found CURL: /usr/lib/libcurl.so (found version "8.9.1")
  -- Found Taglib: /usr/lib/libtag.so (found suitable version "2.0.2", minimum required is "1.12")
  -- Found LibMagic: /usr/lib/libmagic.so
  -- Checking for modules 'libavformat;libavutil;libavcodec'
  --   Found libavformat, version 61.1.100
  --   Found libavutil, version 59.8.100
  --   Found libavcodec, version 61.3.100
  -- Found FFMpegThumbnailer: /usr/lib/libffmpegthumbnailer.so
  -- Found EXIF: /usr/lib/libexif.so (found version "0.6.24")
  -- Found LibExiv2: /usr/lib/libexiv2.so (found version "0.28.3")
  -- Found EBML: /usr/lib/libebml.so
  -- Found Matroska: /usr/lib/libmatroska.so
  -- Found ZLIB: /usr/lib/libz.so (found version "1.3.1")
  -- Configuring unit tests
  -- Found GTest: /usr/lib/cmake/GTest/GTestConfig.cmake (found version "1.15.2")
  -- Configuring done (2.3s)
  -- Generating done (0.0s)
  -- Build files have been written to: /var/cache/makepkg/build/gerbera-nosystemd-git/src/gerbera
  ```
* Note that I also get the following warnings:  
  ```
  [...]
  [ 16%] Building CXX object CMakeFiles/libgerbera.dir/src/config/setup/config_setup_transcoding.cc.o
  /var/cache/makepkg/build/gerbera-nosystemd-git/src/gerbera/src/config/setup/config_setup_transcoding.cc: In static member function ‘static bool ConfigTranscodingSetup::createOptionFromNode(const pugi::xml_node&, std::shared_ptr<TranscodingProfileList>&)’:
  /var/cache/makepkg/build/gerbera-nosystemd-git/src/gerbera/src/config/setup/config_setup_transcoding.cc:129:62: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
    129 |                         prof->addTargetMimeProperty(std::move(TranscodingMimeProperty(key, resource, metadata)));
        |                                                     ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /var/cache/makepkg/build/gerbera-nosystemd-git/src/gerbera/src/config/setup/config_setup_transcoding.cc:129:62: note: remove ‘std::move’ call
  [ 16%] Building CXX object CMakeFiles/libgerbera.dir/src/config/setup/config_setup_tweak.cc.o
  [...]
  [ 42%] Building CXX object CMakeFiles/libgerbera.dir/src/metadata/ffmpeg_handler.cc.o
  /var/cache/makepkg/build/gerbera-nosystemd-git/src/gerbera/src/metadata/ffmpeg_handler.cc: In static member function ‘static void FfmpegHandler::addFfmpegResourceFields(const std::shared_ptr<CdsItem>&, const AVFormatContext*)’:
  /var/cache/makepkg/build/gerbera-nosystemd-git/src/gerbera/src/metadata/ffmpeg_handler.cc:277:65: warning: ‘uint8_t* av_stream_get_side_data(const AVStream*, AVPacketSideDataType, size_t*)’ is deprecated [-Wdeprecated-declarations]
    277 |                     auto displaymatrix = av_stream_get_side_data(st, AV_PKT_DATA_DISPLAYMATRIX, nullptr);
        |                                          ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  In file included from /var/cache/makepkg/build/gerbera-nosystemd-git/src/gerbera/src/metadata/ffmpeg_handler.cc:61:
  /usr/include/libavformat/avformat.h:2126:10: note: declared here
  2126 | uint8_t *av_stream_get_side_data(const AVStream *stream,
        |          ^~~~~~~~~~~~~~~~~~~~~~~
  [ 42%] Building CXX object CMakeFiles/libgerbera.dir/src/metadata/ffmpeg_thumbnailer_handler.cc.o
  [...]
  ```

Regards!

--- a/src/upnp/compat.h
+++ b/src/upnp/compat.h
@@ -44,7 +44,7 @@
 #define UPNP_NEEDS_LITERAL_HOST_REDIRECT
 extern "C" void UpnpSetAllowLiteralHostRedirection(int);
 
-#if (UPNP_VERSION <= 60102)
+#if (UPNP_VERSION <= 60103)
 // new method added
 #define UPNP_NEEDS_CORS
 extern "C" int UpnpSetWebServerCorsString(const char*);

